spec/models/weight_limits_spec.rb

```diff
@@ -91,6 +101,171 @@ def build_inventory_unit(variant, order)
       Spree::ActiveShipping::Config.set(:default_weight => 1)
     end
 
+    describe ".valid_weight_for_package?" do
```

Comment:
I understand your point, though I have some concerns : there are multiple levels of indirection between all the functions being called and the behavior is not consistent (sometimes it's rescued, sometimes it's not ; some valid paths are crashing because of bugs in the code related to nil weight or max weight limit, some paths are not used at all), so it is hard to test all the possible edge cases from the compute or similar higher level methods, and testing each of those invidually is helpful in finding the underlying bugs.

Maybe those tests does not need to be merged but I think they are really useful in understanding the messy parts of the code and documenting current behavior before it can be extracted.

As for the location of the tests file, you're also right. I didn't concern myself with it too much because I was planning on merging the two files once I was finished, and the namespace should be corrected.

Maybe I can just extract it into a new class for now without changing any of the code? It will be still be easier to test and both specs will be leaner and more concise.

Either way, thanks for the feedback.